DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -67,39 +66,65 @@ function Set-TargetResource
         [string]
         $Params,
         [ValidateNotNullOrEmpty()]
-        [string]
+        [string[]]
         $Version,
         [string]
         $Source,
-        [String]
+        [string]
         $chocoParams,
         [bool]
         $AutoUpgrade = $false
     )
     Write-Verbose -Message 'Start Set-TargetResource'
+    $env:Path = [Environment]::GetEnvironmentVariable('Path','Machine')
+    Write-Verbose -Message "Path variables: $env:Path"
 
     if (-Not (Test-ChocoInstalled)) {
         throw "cChocoPackageInstall requires Chocolatey to be installed, consider using cChocoInstaller with 'dependson' in dsc config"
     }
 
-    $isInstalled = IsPackageInstalled -pName $Name
+    if ($Version) {
+        $isInstalled = $true
+        $Version | ForEach-Object {$isInstalled = $isInstalled -and $(IsPackageInstalled -pName $Name -pVersion $_)}
+    } else {
+        $isInstalled = IsPackageInstalled -pName $Name
+    }
+
+    $isInstalledType = IsPackageInstalled -pName $Name
 
     #Uninstall if Ensure is set to absent and the package is installed
-    if ($isInstalled) {
+    if ($isInstalledType) {
         if ($Ensure -eq 'Absent') {
             $whatIfShouldProcess = $pscmdlet.ShouldProcess("$Name", 'Remove Chocolatey package')
             if ($whatIfShouldProcess) {
                 Write-Verbose -Message "Removing $Name as ensure is set to absent"
                 UninstallPackage -pName $Name -pParams $Params
             }
-        } else {
+        } elseif (-not $isInstalled) {

```

Comment:
I'm not sure I understand what is going on here. `$isInstalledType` and `$isInstalled` are both the same value (as they both have the output from `IsPackageInstalled`).

That being the case then the `elseif` will never reached?